090-integrate-bridge-with-metrics-reporter.md

```diff
@@ -0,0 +1,149 @@
+kafka.prometheus.metrics.reporter.listener.enable=false
+```
+
+The MetricsReporter class will be updated to also include a new `StrimziCollectorRegistry` that will work similarly to the `JmxCollectorRegistry`.
+Creating a collector registry for each metrics type ensure better isolation, and makes it easier to add or remove metrics types.
+The `StrimziCollectorRegistry` will include a reference to `PrometheusRegistry.defaultRegistry`, which is the same instance used by the *Reporter* to collect metrics.
+
+The Kafka Bridge will try to load a custom Exporter's configuration file from the path specified by the `bridge.metrics.exporter.config.path` property.
+If the property is not specified or the file is not found, the Kafka Bridge will fall back to the hard-coded configuration file.
+This feature is not strictly required to support the *Reporter*, but will be used by the Cluster Operator.
+
+### Cluster Operator integration
+
+The `KafkaBridge` CRD will also support `metricsConfig` with the addition of `strimziMetricsReporter` type.
+This is how the *Reporter* configuration will look like:
+
+```yaml
+spec:
+  metricsConfig:
+    type: strimziMetricsReporter
+    values:
+      allowList:
+        - "kafka_consumer_consumer_metrics.*"
+        - "kafka_producer_kafka_metrics_count_count"
+        - "kafka_producer_producer_metrics.*"
+```
+
+If the user does not specify the allow list, the default values shown in the previous example will be applied.
+The `KafkaBridgeConfigurationBuilder` will be updated to add metrics configuration when enabled through the `KafkaBridge` resource.
+
+The *Exporter* user provided configuration file will be stored in a ConfigMap and mounted in the Bridge's container.
+The file path passed to the Bridge's container using the `bridge.metrics.exporter.config.path` property will be `/opt/strimzi/custom-config/metrics-config.yml`.
+
+Metrics will be exposed through the Kafka Bridge's HTTP server, so the listener of the *Reporter* will be disabled.
+Other configurations will be locked down with the exception of the `prometheus.metrics.reporter.allowlist` property.
+
+## Affected/not affected projects
+
+The affected projects are Cluster Operator and Kafka Bridge.
+
+## Compatibility
+
+### Bridge
+
+The `KAFKA_BRIDGE_METRICS_ENABLED` environment variable will be deprecated in the next release, and removed in the first major or minor release of 2026.
+When set, the user will get a warning suggesting to use the `bridge.metrics` property.
+In case they are both set, `bridge.metrics` will take precedence over `KAFKA_BRIDGE_METRICS_ENABLED`.
+
+#### Migration
+
+When `KAFKA_BRIDGE_METRICS_ENABLED` will be removed, *Exporter* users will have to set `bridge.metrics=jmxPrometheusExporter` in their `application.properties`.
+
+### Cluster Operator
+
+The `enableMetrics` property in `KafkaBridge` CRD will be deprecated in the next release and removed with the release of Strimzi v1 API.
```

Comment:
If this proposal is accepted we should make sure to raise an issue to remove the `enableMetrics` property when we release the Strimzi v1 API so this isn't forgotten

Reply:
The `enableMetrics` property will be marked as deprecated, so I guess it will be automatically eligible for removal in v1.

Reply:
Yeah - whoever works on the `v1` API would be expected to remove all the deprecated APIs.